src/main/java/hudson/plugins/active_directory/ActiveDirectoryUnixAuthenticationProvider.java

```diff
@@ -436,9 +436,19 @@ private String getPrincipalName(String username, String domainName) {
                 switch (groupLookupStrategy) {
                 case AUTO:
                     // try the accurate one first, and if it's too slow fall back to recursive in the hope that it's faster
-                    long start = System.currentTimeMillis();
-                    boolean found = chainGroupLookup(domainDN, userDN, context, groups);
-                    long duration = (System.currentTimeMillis() - start) / TimeUnit2.SECONDS.toMillis(1);
+                    long start = System.nanoTime();
+                    boolean found = false;
+                    long duration;
+                    duration = TimeUnit2.NANOSECONDS.toSeconds(System.nanoTime() - start);
```

Comment:
duration will be exceptionally small here - as you have not called the LDAP server yet!